pkg/model/openstackmodel/servergroup.go

```diff
@@ -164,6 +161,9 @@ func (b *ServerGroupModelBuilder) buildInstances(c *fi.ModelBuilderContext, sg *
 				}
 				c.AddTask(t)
 			case kops.InstanceGroupRoleMaster:
+				if b.Cluster.Spec.Topology != nil && b.Cluster.Spec.Topology.Masters == kops.TopologyPrivate {
+					continue
+				}
```

Comment:
This looks like a different concern. Perhaps it should be in a different commit?

Reply:
Yep. Done.